src/Microsoft.Identity.Abstractions/DownstreamRestApi/AuthorizationHeaderProviderOptions.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Net.Http;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Options passed-in to call downstream web APIs. To call Microsoft Graph, see rather
+    /// <c>MicrosoftGraphOptions</c> in the <c>Microsoft.Identity.Web.MicrosoftGraph</c> assembly.
+    /// </summary>
+    public class AuthorizationHeaderProviderOptions
+    {
+        AcquireTokenOptions _acquireTokenOptions = new();
+        HttpMethod _httpMethod = HttpMethod.Get;
+        string _protocolScheme = "Bearer";
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        public AuthorizationHeaderProviderOptions()
+        {
+        }
+
+        /// <summary>
+        /// Copy constructor for <see cref="DownstreamRestApiOptions"/>
+        /// </summary>
+        /// <param name="other">Options to copy from.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="other"/> is <c>null</c>.</exception>
+        public AuthorizationHeaderProviderOptions(AuthorizationHeaderProviderOptions other)
+        {
+            if (other == null)
```

Comment:
we are starting to use this pattern for checking for nulls
_ = other ?? throw ...